### ISSUE TITLE
Unable to load PGN from chess24.com due to line character

### ISSUE THREAD

**Issue by yanongena** [jhlywa/chess.js#95]
I download a PGN file from chess24.com that contains my games and when I parse one of these games using chess.load_pgn() it doesn't work. 

After some research, I believe the issue is with the newline character however I cannot the exact problem... 

After some debugging, I found that the header_string var contains the entire PGN string and not just the headers. 

This code:
var newline_char = (typeof options === 'object' &&
            typeof options.newline_char === 'string') ?
                options.newline_char : '\r?\n';
            var regex = new RegExp('^(\[(.|' + mask(newline_char) + ')_\])' +
            '(' + mask(newline_char) + ')_' +
            '1.(' + mask(newline_char) + '|.)*$', 'g');

```
        /* get header part of the PGN file */
        var header_string = pgn.replace(regex, '$1');
```

Basically returns the entire PGN including the moves. I was trying to play with the newline character to see if I can fix it but I cannot seem to find the problem. 

After that, I did some testing and created a workaround that seems to work: 
I remove the header myself and create a string with a single line that contains the moves and pass that to chess.js. It's not nice but it works... 

var lines = game.match(/[^\r\n]+/g);
    var moveLines = "";

```
for (var i = 0; i < lines.length; i++) {
    var line = lines[i];
    if (line.lastIndexOf('"]') > 0) {
        //this is header line
    }
    else {
        moveLines += line;
    }
}
game = moveLines;
```

chess.load_pgn(game);

I attached the file that has been generated by chess24.com

[passero24.txt](https://github.com/jhlywa/chess.js/files/60655/passero24.txt)

**Comment by jhlywa** [comment]
Looks like they're alternating between Unix and Windows line endings.  I'll look into this later this week.  Someone should let chess24 know that they produce crap PGN.

**Comment by yanongena** [comment]
I agree... They also make the mistake with the ambiguous moves but then again, apps like Fritz, Chessbase and others don't have issues with these PGN's.... 

Also, why does my 
var lines = game.match(/[^\r\n]+/g);
work just fine. How is this different from your newline character? \r\n doesn't care if it's windows or Unix, right?

**Comment by jhlywa** [comment]
Yes, chess.js should handle these PGN's too.  The original 3-line regex is a bit cumbersome, I'll try to clean things up over the next few days.

**Comment by neofight78** [comment]
The most recent version of chess.js handles these pgns just fine.
